api/src/main/java/jakarta/enterprise/inject/build/compatible/spi/InvokerFactory.java

```diff
@@ -0,0 +1,24 @@
+package jakarta.enterprise.inject.build.compatible.spi;
+
+import jakarta.enterprise.invoke.InvokerBuilder;
+import jakarta.enterprise.lang.model.declarations.MethodInfo;
+
+/**
+ * Factory for {@link InvokerBuilder}s.
+ *
+ * @since 4.1
+ */
+public interface InvokerFactory {
```

Comment:
I have to say I am not a big fan of this.
You still need `BeanInfo` as an argument, so this just adds an extra step along the way where your `@Registration` method needs an extra argument.
Technically, you could also attempt to pass in some custom impl of `BeanInfo` which I don't suppose we want users to try.

Reply:
Given that a `@Registration` method must declare exactly one parameter of type `BeanInfo` or `InterceptorInfo` or `ObserverInfo`, I considered specifying that an `InvokerFactory` parameter may be declared on a `@Registration` method that declares a parameter of type `BeanInfo` and that `BeanInfo` is the target bean of the invokers created by the factory, and then `InvokerFactory.createInvoker()` would only have a single parameter of type `MethodInfo`. In the end, it felt a little too magical to me.

The main reason why I don't want `BeanInfo.createInvoker()` is that `BeanInfo` (like all other `*Info` types) is a "passive" object that only hands out information. It shouldn't have any "active" methods that configure behavior. 

An alternative would be something like

```java
/**
 * Allows configuring a bean.
 *
 * @see Registration
 * @since 4.1
 */
public interface BeanConfig {
    /**
     * Returns the {@link BeanInfo} corresponding to this configured bean.
     *
     * @return the {@link BeanInfo} corresponding to this configured bean, never {@code null}
     */
    BeanInfo info();

    /**
     * Returns a new {@link InvokerBuilder} for given method of this bean. The builder
     * eventually produces an opaque representation of the generated invoker.
     * <p>
     * If an invoker may not be built for this bean or for given {@code method},
     * an exception is thrown.
     *
     * @param method target method of the invoker, must not be {@code null}
     * @return the invoker builder, never {@code null}
     */
    InvokerBuilder<InvokerInfo> createInvoker(MethodInfo targetMethod);
}
```

I'm a little hesitant about this, because it would also be a natural place for other bean configurations (bean attribute transformations, injection point transformations) and I have doubts we can do that in the `@Registration` phase, as currently specified.